cmake/Helpers.cmake

```diff
@@ -0,0 +1,790 @@
+FUNCTION(SET_INTERNAL VAR_NAME VALUE)
+    SET(${VAR_NAME} ${VALUE} CACHE INTERNAL "")
+ENDFUNCTION()
+
+FUNCTION(ADD_FLAG_IF_SUPPORTED)
+    SET(options "")
+    SET(oneValueArgs FLAG TEST_VALUE VALUE)
+    SET(multiValueArgs "")
+    CMAKE_PARSE_ARGUMENTS("" "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    IF(${_TEST_VALUE})
+        SET(TEST_FLAG "${_FLAG}=${_TEST_VALUE}")
+    ELSEIF(${_VALUE})
+    SET(TEST_FLAG "${_FLAG}=${_VALUE}")
+    ELSE ()
+        SET(TEST_FLAG "${_FLAG}")
+    ENDIF ()
+    
+    CHECK_CXX_COMPILER_FLAG("${TEST_FLAG}" SUPPORTED)
+
+    IF(SUPPORTED)
+        IF(${_VALUE})
+            SET(_FLAG " ${_FLAG}=${_VALUE}")
+        ENDIF ()
+        STRING(APPEND CMAKE_CXX_FLAGS "${_FLAG}")
+    ENDIF ()
+ENDFUNCTION(ADD_FLAG_IF_SUPPORTED)
+
+FUNCTION(CONFIGURE_COMPILER)
+    INCLUDE(CheckCXXCompilerFlag)
+    IF(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+        IF(ENABLE_WERROR)
+            STRING(JOIN " "
+                CMAKE_CXX_FLAGS
+                ${CMAKE_CXX_FLAGS}
+                "-Werror"
+                "-Wno-inline"
+                "-Wno-error=cast-align"
+                "-Wmissing-declarations"
+            )
+        ELSE ()
+            ADD_FLAG_IF_SUPPORTED(
+                FLAG "-ffile-prefix-map"
+                TEST_VALUE "a=b"
+                VALUE "${CMAKE_SOURCE_DIR}=."
+            )		
+        ENDIF ()
+
+        STRING(JOIN " "
+            CMAKE_CXX_FLAGS
+            ${CMAKE_CXX_FLAGS}
+            "-Wall"
+            "-W"
+            "-Wpointer-arith"
+            "-Wno-format-y2k"
+            "-Wno-long-long"
+        )
+
+        ADD_FLAG_IF_SUPPORTED(FLAG "-pedantic")
+        # mark chars explicitly signed (ARM defaults to unsigned)
+        ADD_FLAG_IF_SUPPORTED(FLAG "-fsigned-char")
+        # only export symbols explicitly marked to be exported.
+        ADD_FLAG_IF_SUPPORTED(FLAG "-fvisibility=hidden")
+        # Fast math helps us covering some things that need a little more rework soon
+        ADD_FLAG_IF_SUPPORTED(FLAG "-ffast-math")
+        ADD_FLAG_IF_SUPPORTED(FLAG "-frounding-math")
+
+        IF(WIN32)
+            # GCC 4.5.0+ has shared libstdc++ without dllimport
+            STRING(APPEND CMAKE_SHARED_LINKER_FLAGS " -Wl,--enable-auto-import")
+            STRING(APPEND CMAKE_MODULE_LINKER_FLAGS " -Wl,--enable-auto-import")
+        ENDIF (WIN32)
+        # Ensure all plugin symbols exist.
+        IF(NOT APPLE AND NOT UNSAFE_PLUGIN)
+            string(APPEND CMAKE_MODULE_LINKER_FLAGS " -Wl,--no-undefined")
+        ENDIF (NOT APPLE AND NOT UNSAFE_PLUGIN)
+    ENDIF ()
+
+    IF(CMAKE_COMPILER_ID MATCHES "GNU")
+        IF(ENABLE_WERROR)
```

Comment:
This check is redundant.

Reply:
I just only rearanged original code:

https://github.com/gemrb/gemrb/blob/82b1b9cbbb9c5866e2d0cab15496221036cddfb1/CMakeLists.txt#L220

I wanted to make it like this:

```
if(GNU | Clang)
 //common stuff
endif()

if(GNU)
 //GNU specific
endif()
```

Reply:
-Wno-error subflags will be noops if -Werror is not enabled, so there's no harm in always adding them (that I know of).